packages/grid/_modules_/grid/hooks/utils/useGridLogger.ts

```diff
@@ -39,35 +41,35 @@ function getAppender(name: string, logLevel: string, appender: Logger = console)
   return logger as Logger;
 }
 
-const defaultFactory: (logLevel: string) => LoggerFactoryFn =
-  (logLevel: string) => (name: string) =>
-    getAppender(name, logLevel);
-
-type LoggerFactoryFn = (name: string) => Logger;
-
-// TODO Refactor to allow different logger for each grid in a page...
-let factory: LoggerFactoryFn | null;
-
-export function useLoggerFactory(
-  apiRef: any,
+export function useGridLoggerFactory(
+  apiRef: GridApiRef,
   props: Pick<GridComponentProps, 'logger' | 'logLevel'>,
 ) {
-  if (forceDebug) {
-    factory = defaultFactory('debug');
-    return;
-  }
+  apiRef.current.getLogger = React.useCallback<GridLoggerApi['getLogger']>(
```

Comment:
I can't use `useGridApiMethod` because it's using `useGridLogger` so it's initialized after.
We can either 

- remove the logger usage in `useGridApiMethod`

- add manually the method to `apiRef` (like in this PR)

- do not use `apiRef` to store the callback (don't like this one)

```ts
const loggers: { [gridId: number]: LoggerFactoryFn } = {}

const useGridLoggerFactory = (apiRef, props) => {
  const id = useUniqId()

  if (!loggers[id]) {
    loggers[id] = createLogger()
    apiRef.id = id
  }
}

const useGridLogger = (apiRef, name) => {
  const getLogger = loggers[apiRef.id]
  ...
}
``` 

- create the logger in a component above `GridComponent` and expose it through a React context. This would add a React component to the tree but would allow to reuse the same logger system for other components.